internal/service/accesslistapikey/data_source_accesslist_api_keys.go

```diff
@@ -68,27 +70,44 @@ func PluralDataSource() *schema.Resource {
 	}
 }
 
-func dataSourceMongoDBAtlasAccessListAPIKeysRead(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
-	// Get client connection.
-	conn := meta.(*config.MongoDBClient).Atlas
-	options := &matlas.ListOptions{
-		PageNum:      d.Get("page_num").(int),
-		ItemsPerPage: d.Get("items_per_page").(int),
-	}
-
+func dataSourcePluralRead(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	connV2 := meta.(*config.MongoDBClient).AtlasV2
 	orgID := d.Get("org_id").(string)
 	apiKeyID := d.Get("api_key_id").(string)
-
-	accessListAPIKeys, _, err := conn.AccessListAPIKeys.List(ctx, orgID, apiKeyID, options)
+	params := &admin.ListApiKeyAccessListsEntriesApiParams{
+		PageNum:      pointy.Int(d.Get("page_num").(int)),
+		ItemsPerPage: pointy.Int(d.Get("items_per_page").(int)),
```

Comment:
Since this now defines a pointer to the int value of the attribute, would there be a risk we send items per page to 0 if this attribute is not defined?

Reply:
if the attribute is not sent, the API should apply the default value (`0`) anyway

Reply:
you're totally right!

Reply:
Just to clarify the risk is in ItemsPerPage which has a default value of 100, if a pointer value to 0 is defined the response will return 0 items when the attribute is not defined.

Reply:
ah thanks 👍 , I thought we were talking about `page_num`

Reply:
it was working with items_per_page = 0, i suppose the server was assuming the default value.   anyway it's fixed now to send nil if values are 0